angular/src/app/services/shopping-cart.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { IBook } from '../interfaces/global.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ShoppingCartService {
+  public booksSubject = new Subject<IBook[]>();
+  books: IBook[] = [];
+
+  setBook(book: IBook) {
+    this.books.push(book);
+    this.booksSubject.next(this.books);
+  }
+
+  getBooks() {
+    return this.booksSubject;
+  }
+
+  removeBook(id: number) {
+    this.books = this.books.filter((book) => book.id !== id);
```

Comment:
Yoni, you don't need to create the books variable if you can iterate what you get from getBooks

Reply:
Hello migue I was trying to carry out this process but for some unknown reason I cannot obtain the values ​​of said function I tried to carry out with a behaviorsubject and it did not work either.